pkgs/data/misc/cacert/default.nix

```diff
@@ -1,123 +1,211 @@
-    ];
-    sha256 = "1d4q27j1gss0186a5m8bs5dk786w07ccyq0qi6xmd2zr1a8q16wy";
+  isPathIntoStore = f: builtins.isString f && lib.substring 0 (lib.stringLength builtins.storeDir) f == builtins.storeDir;
+  extraCertificateFiles = lib.filter (f: builtins.isPath f || isPathIntoStore f) extraCertificates;
+  extraCertificateStrings = lib.filter builtins.isString extraCertificates;
+
+  # Build a cut-down version of certdata.
+  # This allows people to specify a blocklist/extra certificates without downloading the entire NSS source code,
+  # and allows substituting of just the certdata portion from e.g. cache.nixos.org.
+  nssCertdata = stdenv.mkDerivation rec {
+    pname = "nss-certdata";
+    version = "3.66";
+
+    src = fetchurl {
+      url = "mirror://mozilla/security/nss/releases/NSS_${lib.replaceStrings ["."] ["_"] version}_RTM/src/nss-${version}.tar.gz";
+      sha256 = "1jfdnh5l4k57r2vb07s06hqi7m2qzk0d9x25lsdsrw3cflx9x9w9";
+    };
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir -p $out/nss/lib/ckfw/builtins
+      cp nss/lib/ckfw/builtins/certdata.txt $out/nss/lib/ckfw/builtins/certdata.txt
+    '';
   };
 in
 
 stdenv.mkDerivation rec {
   pname = "nss-cacert";
-  version = "3.66";
+  version = if nssOverride != null then nssOverride.version else nssCertdata.version;
 
-  src = fetchurl {
-    url = "mirror://mozilla/security/nss/releases/NSS_${lib.replaceStrings ["."] ["_"] version}_RTM/src/nss-${version}.tar.gz";
-    sha256 = "1jfdnh5l4k57r2vb07s06hqi7m2qzk0d9x25lsdsrw3cflx9x9w9";
-  };
+  src = if nssOverride != null then nssOverride.src else nssCertdata;
 
-  outputs = [ "out" "unbundled" ];
+  outputs = [ "out" "unbundled" "p11kit" ];
 
-  nativeBuildInputs = [ python3 ];
+  nativeBuildInputs = [ buildcatrust ];
 
   configurePhase = ''
     ln -s nss/lib/ckfw/builtins/certdata.txt
 
-    cat << EOF > blacklist.txt
-    ${lib.concatStringsSep "\n" (map (c: ''"${c}"'') blacklist)}
+    cat << EOF > blocklist.txt
+    ${lib.concatStringsSep "\n" blacklist}
     EOF
 
-    # copy from the store, otherwise python will scan it for imports
-    cat "${certdata2pem}" > certdata2pem.py
+    cat << EOF > extra-certificates-bundle.crt
+    ${lib.concatStringsSep "\n\n" extraCertificateStrings}
+    EOF
   '';
 
   buildPhase = ''
-    python certdata2pem.py | grep -vE '^(!|UNTRUSTED)'
-
-    for cert in *.crt; do
-      echo $cert | cut -d. -f1 | sed -e 's,_, ,g' >> ca-bundle.crt
-      cat $cert >> ca-bundle.crt
-      echo >> ca-bundle.crt
-    done
+    mkdir unbundled
+    buildcatrust \
+      --certdata_input certdata.txt \
+      --ca_bundle_input extra-certificates-bundle.crt ${lib.concatStringsSep " " extraCertificateFiles} \
+      --blocklist blocklist.txt \
+      --ca_bundle_output ca-bundle.crt \
+      --ca_unpacked_output unbundled \
+      --p11kit_output ca-bundle.trust.p11-kit
   '';
 
   installPhase = ''
```

Comment:
A bit surprised that nothing here uses `install(1)`, but eh.

Reply:
Now it does. Much nicer! :P